src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -83,33 +82,99 @@ def map_tensor_names_to_types(input: ConstOutput) -> dict:
 
 
 class InferRequest(InferRequestBase):
-    """InferRequest wrapper."""
+    """
+    InferRequest class represents infer request which can be run in
+    asynchronous or synchronous manners.
+    """
 
     def infer(self, inputs: Union[dict, list] = None) -> dict:
-        """Infer wrapper for InferRequest."""
+        """
+        Infers specified input(s) in synchronous mode.
+        Blocks all methods of InferRequest while request is running.
+        Calling any method will lead to throwning exceptions.
+
+        Parameters
+        ----------
+        inputs : dict[Union[int, str, openvino.runtime.ConstOutput] : openvino.runtime.Tensor]
+            Data to set on input tensors.
+
+        Returns
+        ----------
+        infer : dict[openvino.runtime.ConstOutput : openvino.runtime.Tensor]
+            Dictionary of results from output tensors with ports as keys.
+        """
         return super().infer(
             {} if inputs is None else normalize_inputs(inputs, get_input_types(self))
         )
 
     def start_async(
         self, inputs: Union[dict, list] = None, userdata: Any = None
     ) -> None:
-        """Asynchronous infer wrapper for InferRequest."""
+        """
+        Starts inference of specified input(s) in asynchronous mode.
+        Returns immediately. Inference starts also immediately.
+        Calling any method while the request is running will lead to
+        throwning exceptions.
+
+        Parameters
+        ----------
+        inputs : dict[Union[int, str, openvino.runtime.ConstOutput] : openvino.runtime.Tensor]
+            Data to set on input tensors.
+
+        userdata : Any
+            Any data that will be passed inside callback call.                
+
+        Returns
+        ----------
+        start_async : None
+        """
         super().start_async(
             {} if inputs is None else normalize_inputs(inputs, get_input_types(self)),
             userdata,
         )
 
 
 class CompiledModel(CompiledModelBase):
-    """CompiledModel wrapper."""
+    """
+    CompiledModel represents Model that is compiled for a specific device by applying
+    multiple optimization transformations, then mapping to compute kernels.
+    """
 
     def create_infer_request(self) -> InferRequest:
-        """Create new InferRequest object."""
+        """
+        Creates an inference request object used to infer the compiled model.
+        The created request has allocated input and output tensors.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        ----------
+        create_infer_request : openvino.runtime.InferRequest
+            New InferRequest object.
+        """
         return InferRequest(super().create_infer_request())
 
     def infer_new_request(self, inputs: Union[dict, list] = None) -> dict:
-        """Infer wrapper for CompiledModel."""
+        """
+        Infers specified input(s) in synchronous mode.
```

Comment:
Shouldn't we add a description for this https://github.com/openvinotoolkit/openvino/blob/5edf9fad4dd582d13258d4ea4b541b1ecdbaee41/src/bindings/python/src/openvino/runtime/ie_api.py#L183? It's not clear that is works the same as `infer_new_request`

Reply:
Will add that this is the same as calling `infer_new_request` and to look at it for reference.